docs/specs/pipelines-io.md

```diff
@@ -0,0 +1,196 @@
+## Efficient IO
+
+As part of the high performance work, we need new abstractions for doing efficient reading and writing to and from the network. The pattern we have settled on
```

Comment:
Consider using I/O in general here, with networking as a specific example.  Pipelines should be applicable to all types of IO: file, IPC and network.